src/test/java/io/javalin/TestWebSocket.kt

```diff
@@ -78,18 +78,20 @@ class TestWebSocket {
 
     @Test
     fun `general integration test`() = TestUtil.test(contextPathJavalin) { app, _ ->
-        val userUsernameMap = LinkedHashMap<WsSession, Int>()
+        val contextByIdMap = LinkedHashMap<String, WsContext>()
```

Comment:
Notice, since we don't delegate to `hashCode` and `equals` anymore, and WsContext is a new instance everytime, and won't be present in the map. In the pull request we use WsContext.id as alternative identifier.

Some of options:
1. We're ok as it's done in this pull request.
2. Let's continue delegating to `hashCode` and `equals` of a real ws session, and allow confusions such as: wsContext == wsSession
3. Somehow keep WsContext instances, just as jetty does with ws sessions?

Reply:
I like leaning on Jetty, so my gut feeling is that 2. is the safest. Can you think of any cases where `wsContext == wsSession` would come up/be confusing?

Reply:
Probably something makes me feel suspicious, when I compare objects of different types and they can be equal (not speaking of null case). Ignoring this feeling, and accepting the idea that context ~== session, the equality isn't that bad. I'll apply `2.` method to the pull request.

Reply:
For me it makes sense that two `WsContex` objects should be equal if they're built from the same `Session`. I agree that it'd be weird if you directly compared a `Session` and a `WsContext`, but I don't think that's likely to happen, it's more of a hidden implementation detail.